hazelcast/src/main/java/com/hazelcast/internal/cluster/impl/ClusterJoinManager.java

```diff
@@ -858,7 +821,55 @@ && isMemberRejoining(memberMap, member.getAddress(), member.getUuid())) {
         }
     }
 
-    private void runProvidedPostJoinOpsWithoutBroadcast() {
+    private boolean shouldTriggerRepartitionSyncStrategyOnly(MemberMap memberMap) {
+        if (syncJoinStrategy != null) {
```

Comment:
This is the culprit that breaks tests for delayed rebalancing when persistence is enabled on Hazelcast Enterprise side. The checks to determine if it's a rejoining member with persistence are completely omitted when `syncJoinStrategy == null`.

@lprimak is there any reasoning for that omission? I anyway attach a patch that restores the check (the previously failing EE tests pass with that patch applied).

[clusterJoinManager.patch](https://github.com/hazelcast/hazelcast/files/12883206/clusterJoinManager.patch)

Reply:
Ah, I was wondering the same thing. The only thing I can think of is that the persistence code appeared after the initial PR was done years ago... in any case, the applied patch looks good so LGTM